Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Program.cs

```diff
@@ -0,0 +1,153 @@
+            var platform = new ClientPlatform(platformSettings, m_logger);
+
+            // Prepare endpoint
+            var endpointSettings = new ApplicationEndpointSettings(new SipUri(QuickSamplesConfig.ApplicationEndpointId));
+            var applicationEndpoint = new ApplicationEndpoint(platform, endpointSettings, EventChannel);
+
+            var loggingContext = new LoggingContext(Guid.NewGuid());
+            await applicationEndpoint.InitializeAsync(loggingContext).ConfigureAwait(false);
+            await applicationEndpoint.InitializeApplicationAsync(loggingContext).ConfigureAwait(false);
+
+            // Meeting configuration
+            var meetingConfiguration = new AdhocMeetingCreationInput(Guid.NewGuid().ToString("N") + " test meeting");
+
+            // Schedule meeting
+            var adhocMeeting = await applicationEndpoint.Application.CreateAdhocMeetingAsync(loggingContext, meetingConfiguration).ConfigureAwait(false);
+
+            WriteToConsoleInColor("ad hoc meeting uri : " + adhocMeeting.OnlineMeetingUri);
+            WriteToConsoleInColor("ad hoc meeting join url : " + adhocMeeting.JoinUrl);
+
+            // Get all the events related to join meeting through Trouter's uri
+            platformSettings.SetCustomizedCallbackurl(new Uri(EventChannel.CallbackUri));
+
+            // Start joining the meeting
+            var invitation = await adhocMeeting.JoinAdhocMeeting(loggingContext, null).ConfigureAwait(false);
+
+            // Wait for the join to complete
+            await invitation.WaitForInviteCompleteAsync().ConfigureAwait(false);
+
+            invitation.RelatedConversation.HandleParticipantChange += Conversation_HandleParticipantChange;
+
+            // invite the participant to join the meeting
+            WriteToConsoleInColor("Invite " + participantUri+" to join the meeting");
+            await invitation.RelatedConversation.AddParticipantAsync(participantUri, loggingContext).ConfigureAwait(false);
+           
+            WriteToConsoleInColor("Showing roaster udpates for 5 minutes for meeting : " + adhocMeeting.JoinUrl);
+
+            // Wait 5 minutes before exiting.
+            // Since we have registered Conversation_HandleParticipantChange, we will continue to show participant changes in the
+            // meeting for this duration.
+            await Task.Delay(TimeSpan.FromMinutes(5)).ConfigureAwait(false);
+        }
+
+        private void Conversation_HandleParticipantChange(object sender, ParticipantChangeEventArgs eventArgs)
+        {
+            if (eventArgs.AddedParticipants?.Count > 0)
+            {
+                foreach (var participant in eventArgs.AddedParticipants)
+                {
+                    WriteToConsoleInColor(participant.Name + " has joined the meeting.");
+
+                    var participantUri = ConfigurationManager.AppSettings["ParticipantUri"];
+                    if (participantUri.Equals(participant.Uri))
```

Comment:
Can you please move this logic to RunAsync() method? See my comment above on how you can do that.